libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -97,36 +109,36 @@ int32_t SecureSocketsTransport_Send( const NetworkContext_t * pNetworkContext,
         ( pNetworkContext == NULL ) )
     {
         LogError( ( "Invalid parameter: pMessage=%p, bytesToSend=%lu, pNetworkContext=%p",
-                    pMessage, bytesToSend, pNetworkContext ) );
+                    pMessage, bytesToSend, ( void * ) pNetworkContext ) );
         bytesSent = SOCKETS_EINVAL;
     }
-    else if( ( pNetworkContext != NULL ) && ( pNetworkContext->pContext == NULL ) )
+    else if( ( pNetworkContext != NULL ) && ( pNetworkContext->tcpSocket == SOCKETS_INVALID_SOCKET ) )
     {
-        LogError( ( "Invalid parameter: pNetworkContext->pContext cannot be NULL." ) );
+        LogError( ( "Invalid parameter: pNetworkContext->tcpSocket cannot be SOCKETS_INVALID_SOCKET." ) );
         bytesSent = SOCKETS_EINVAL;
     }
     else
     {
-        bytesSent = SOCKETS_Send( ( Socket_t ) pNetworkContext->pContext,
+        bytesSent = SOCKETS_Send( pNetworkContext->tcpSocket,
                                   pMessage,
                                   bytesToSend,
                                   0 );
 
         /* If an error occurred, a negative value is returned. @ref SocketsErrors. */
         if( bytesSent >= 0 )
         {
-            if( bytesSent < bytesToSend )
+            if( bytesSent < ( int32_t ) bytesToSend )
             {
-                LogWarn( ( "bytesSent %ld < bytesToSend %ld.", bytesSent, bytesToSend ) );
+                LogWarn( ( "bytesSent %d < bytesToSend %d.", bytesSent, bytesToSend ) );
```

Comment:
`bytesToSend` is of `size_t` which requires use of `lu` format specifier
```suggestion
                LogWarn( ( "bytesSent %d < bytesToSend %lu.", bytesSent, bytesToSend ) );
```